### ISSUE TITLE
WebClient + PagedResources Support

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-hateoas#741]
Right now it does not appear that there is `WebClient` and `PagedResources` support. I'd like Hateoas to provide the proper integration and Boot (create a separate ticket if necessary) to integrate this into the default `WebClient.Builder`.

If I try and retrieve `PagedResources` the content is empty even if there is content.

```
ParameterizedTypeReference<PagedResources<Message>> type =
		new ParameterizedTypeReference<PagedResources<Message>>() {};
PagedResources<Message> response = this.client.get()
		.uri(this.url)
		.retrieve()
		.bodyToMono(type)
		.block();
// fails even when content is present
assertThat(response.getContent()).hasSize(1);
```

Here is the body

```json
{
  "_embedded" : {
    "messages" : [ {
      "text" : "Hi",
      "_links" : {
        "self" : {
          "href" : "http://localhost:42101/messages/1"
        },
        "message" : {
          "href" : "http://localhost:42101/messages/1"
        }
      }
    } ]
  },
  "_links" : {
    "self" : {
      "href" : "http://localhost:42101/messages{?page,size,sort}",
      "templated" : true
    },
    "profile" : {
      "href" : "http://localhost:42101/profile/messages"
    }
  },
  "page" : {
    "size" : 20,
    "totalElements" : 1,
    "totalPages" : 1,
    "number" : 0
  }
}
```

Here is `Message`

```java
@Entity
public class Message {
	@Id
	@GeneratedValue
	private Long id;

	private String text;

	public Message() {
	}

	public Message(String text) {
		this.text = text;
	}

	public Long getId() {
		return this.id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getText() {
		return this.text;
	}

	public void setText(String text) {
		this.text = text;
	}
}
```

Finally `MessageRepository`

```java
@RepositoryRestResource
public interface MessageRepository extends PagingAndSortingRepository<Message, Long> {
}
```

Here is a complete sample to demonstrate the issue https://github.com/rwinch/spring-hateoas-webclient/blob/master/src/test/java/sample/springhateoaswebclient/SpringHateoasWebclientApplicationTests.java#L56

**Comment by gregturn** [comment]
@rwinch If you have any guidance on how to interrogate a `WebClient` to discern what message I would appreciate it.

I starting drafting

```java
	@Test
	public void registersHttpMessageConvertersForWebClient() {

		withContext(WebClientHalConfig.class, context -> {

			WebClient client = context.getBean(WebClient.class)
				.mutate()
				.exchangeStrategies(ExchangeStrategies.withDefaults())
				.build();

			assertThat(1).isEqualTo(1);
		});
	}

	@Configuration
	@EnableHypermediaSupport(type = HypermediaType.HAL)
	@EnableWebMvc
	static class WebClientHalConfig {

		@Bean
		public WebClient webClient() {
			return WebClient.create();
		}
	}
```

But..there is little to assert against due to the highly encapsulated nature of `WebClient`. I can see a bunch of info in my debugger.

**Comment by rwinch** [comment]
Sorry...I'm not sure I understand what you are asking? Can you rephrase your question please?

**Comment by rwinch** [comment]
Unfortunately I don't have any alternatives for you. Spring Framework typically points to a `MockWebServer` instance for [their tests](https://github.com/spring-projects/spring-framework/blob/8aa6e5bfea2c7314deaa1b432554e9e914b09ee7/spring-web/src/test/java/org/springframework/http/client/AbstractMockWebServerTestCase.java#L21). This is also what I do for my tests.

**Comment by gregturn** [comment]
Follow the progress of #728.

**Comment by gregturn** [comment]
Resolved via #728.
